service/raft/raft_address_map.hh

```diff
@@ -164,13 +173,90 @@ class raft_address_map {
         }
     }
 
+    template <std::invocable<raft_address_map&> F>
+    void replicate(F f, std::experimental::source_location l = std::experimental::source_location::current()) {
+        if (_gate.is_closed()) {
+            return;
+        }
+
+        _replication_fiber = _replication_fiber.then([this, f = std::move(f), l] () -> future<> {
+            if (_gate.is_closed()) {
+                co_return;
+            }
+
+            auto holder = _gate.hold();
+            try {
+                co_await this->container().invoke_on_others([f] (raft_address_map& self) {
+                    if (self._gate.is_closed()) {
+                        return;
+                    }
+
+                    f(self);
+                });
+            } catch (...) {
+                rslog.error("raft_address_map::replicate (called from {}) failed: {}",
+                            l.function_name(), std::current_exception());
+            }
+        });
+    }
+
+    void replicate_set_nonexpiring_entry(const raft::server_id& id, const gms::inet_address& addr) {
+        replicate([id, addr] (raft_address_map& self) {
+            self.handle_set_nonexpiring_entry(id, addr);
+        });
+    }
+
+    void replicate_set_expiring_flag(const raft::server_id& id) {
+        replicate([id] (raft_address_map& self) {
+            self.handle_set_expiring_flag(id);
+        });
+    }
+
+    void handle_set_nonexpiring_entry(const raft::server_id& id, const gms::inet_address& addr) {
+        auto [it, _] = _map.try_emplace(id, addr);
+        auto& entry = it->second;
+
+        // Don't allow to remap to a different address
```

Comment:
I think this check is usual now that we know where we are heading and especially since you're doing it in the background now it would simply abort the fiber in question. We should expect that the check is done at the call site, if it's  deemed necessary, and replicate() itself doesn't have any bits of the business logic and is solely focused on replicating the entries.

Reply:
My intention for this was to notify the next developer (you :)) that this piece of code needs to be updated when IP change is introduced - hopefully by failing tests (in tests, `on_internal_error` causes node to crash).

I can modify this to handle remapping. But this won't cover IP change support fully - other places in raft_address_map will need a modification as well.

Reply:
You don't need to modify it to handle remapping, you can just drop the check.

Reply:
`try_emplace` does not modify the entry's value if one already exists. So if I simply removed the `if`, the function would not remap.